docs/DeveloperGuide.adoc

```diff
@@ -718,6 +732,52 @@ The added data security will assure the user that their data is safe and allows
 
 // end::dataencryption[]
 
+//tag::ai[]
+=== [Proposed] Artificial Intelligence (AI)
+Artificial intelligence (AI) is the simulation of human intelligence processes by machines, especially computer systems. These processes include learning (the acquisition of information and rules for using the information), reasoning (using rules to reach approximate or definite conclusions) and self-correction.
+We want `**THRIFT**` to possess the ability to make calculated decisions and provide the user with appropriate recommendations under the appropriate situations.
+This way, we can enhance the user experience and make our application more effective.
+
+The following sections will describe how we plan to implement this feature and the possible design considerations.
+
+==== Proposed Implementation
+Adhering to Open-Closed principle, we will not modify the existing architecture but extend it.
+The following class diagram illustrates how we propose to do it, by adding 2 new classes to the existing architecture:
+
+image::ai/AiClassDiagram.png[]
+
+The new classes will have the following roles:
+
+* `**AI**`: Analyses data from the `**Data**` and perform machine learning on it.
+If the need arises, feedback to the `**Data**` class to morph the data inside.
+Modifies the actions taken by the `**LogicManager**` class based on the machine learning model.
+
+* `**Data**`:  Collects data from `**Thrift**` and formats to it to a form understandable by the `**AI**` class.
+It can be modified by the `**AI**` class in the process of machine learning.
+
+==== Design consideration
+
+*Aspect: Portability of the Learning Model*
+
+The portability of the learning model will determine the complexity of the implementation, and the following designs can be considered:
+
+* *Alternative 1*: Include the machine learning algorithms and models with the application
+** Pro: Allows the application to be portable and self-contained
+** Con: Results in larger sizes of the application
```

Comment:
Results in **a** larger **size** of the application

Reply:
This has been rephrased.